File: HubWebApp/Lib/XTI_Hub.Abstractions/AppUserGroupName.cs
using XTI_Core;

namespace XTI_Hub.Abstractions;

public sealed class AppUserGroupName : TextKeyValue, IEquatable<AppUserGroupName>
{
    public static readonly AppUserGroupName XTI = new AppUserGroupName(nameof(XTI));
    public static readonly AppUserGroupName General = new AppUserGroupName(nameof(General));

    public AppUserGroupName()
        : this("")
    {
    }

    public AppUserGroupName(string displayText) : base(displayText)
    {
    }

    public bool Equals(AppUserGroupName? other) => _Equals(other);
}

File: HubWebApp/Apps/HubWebApp/Scripts/Lib/AppUserGroupName.ts

export class AppUserGroupName {
    readonly value: string;
    readonly displayText: string;

    constructor(source?: IAppUserGroupName) {
        this.value = source ? source.Value : "";
        this.displayText = source ? source.DisplayText : "";
    }

    toString() { return this.displayText; }
}